Sources/AppKitNavigation/Navigation/ModalSessionContent.swift

```diff
@@ -0,0 +1,22 @@
+#if canImport(AppKit) && !targetEnvironment(macCatalyst)
+
+import AppKit
+
+@MainActor
+public protocol ModalSessionContent: ModalContent {
+    func appKitNavigationBeginModalSession() -> NSApplication.ModalSession
+}
+
+extension NSWindow: ModalSessionContent {
+
+    public func appKitNavigationBeginModalSession() -> NSApplication.ModalSession {
+        __appKitNavigationBeginModalSession()
+    }
+    
+    @objc func __appKitNavigationBeginModalSession() -> NSApplication.ModalSession {
+        let modalSession = NSApplication.shared.beginModalSession(for: self)
+        return modalSession
+    }
```

Comment:
Because the Modal way of NSWindow is different from the Modal way of NSSavePanel, but NSSavePanel inherits from NSWindow, Swift protocol extensions can't override the methods in the subclasses, this one is not used yet, and it can be deleted for the time being.